scripts/build-upload-gate

```diff
@@ -0,0 +1,43 @@
+!/bin/bash
+set -e
+set -x
+
+BUILD_DEBUG="${BUILD_DEBUG:-}"
+if [[ -n "${BUILD_DEBUG}" ]]; then
+    set -x
+    env
+fi
+
+if [[ -n "$DRONE_TAG" ]]; then
+    echo "Build has been triggered by a tag. Skipping gate"
+    exit 0
+fi 
+
+if [[ -z "$DRONE_BRANCH" ]]; then 
+    echo "No branch detected. Skipping gate"
+    exit 0
+fi
+
+if [[ -z "$DRONE_REPO" ]]; then 
+    echo "No repository detected. Unable to gate"
+    exit 1
+fi
+
+if [[ -z "$DRONE_COMMIT" ]]; then 
+    echo "No commit detected. Unable to gate"
+    exit 1
+fi
+
+echo "Repo: $DRONE_REPO"
+echo "Branch: $DRONE_BRANCH"
+echo "Build Commit: $DRONE_COMMIT"
+
+LATEST_BRANCH_COMMIT=$(curl -s https://api.github.com/repos/$DRONE_REPO/branches/$DRONE_BRANCH | jq -r .commit.sha)
```

Comment:
GitHub's API has a fairly low limit on unauthed requests so this should either pass auth (Nelson can add a secret to drone) or look for failure and just go back to assuming the build should be done.

Reply:
On my local env this is 60 per hour. I've added a check now that will pass the gate if rate limiting has been hit. Think there were some better ways to do it (check `X-RateLimit-Remaining` in the response header but has more lines of script... or always check rate limiting up front via `/rate_limit` but that's an additional request every time although not applicable to limiting). Opted for somewhere in the middle.